numpyro/hmc_util.py

```diff
@@ -543,7 +544,12 @@ def _body_fn(state):
 
 def log_density(model, model_args, model_kwargs, params):
     def logp(d, val):
-        return d.logpdf(val) if isinstance(d.dist, jax_continuous) else d.logpmf(val)
+        if isinstance(d, Distribution):
+            return d.log_prob(val)
+        elif isinstance(d.dist, jax_continuous):
+            return d.logpdf(val)
+        else:
+            return d.logpmf(val)
```

Comment:
I think you can go ahead and drop supporting `jax_continuous/discrete`. We can add a `log_prob` method to `jax_frozen` and still have all versions work.

Reply:
Sounds good. I'll do this separately, with the name changes and file reorg.